runtime/queries/metricsview_aggregation.go

```diff
@@ -305,6 +310,40 @@ func (q *MetricsViewAggregation) rewriteToMetricsViewQuery(export bool) (*metric
 		}
 	}
 
+	// If there is only one time dimension and null fill is enabled, we set the spine to the time range
+	if q.FillMissing && len(computedTimeDims) > 0 {
+		if qry.Spine != nil {
+			// should we silently ignore instead of error ?
+			return nil, fmt.Errorf("cannot have both where and time spine")
+		}
+		if len(computedTimeDims) > 1 {
+			return nil, fmt.Errorf("cannot have multiple time spines")
+		}
+		if q.TimeRange == nil || q.TimeRange.Start == nil || q.TimeRange.End == nil {
+			return nil, fmt.Errorf("time range is required for null fill")
+		}
+
+		timeDim := computedTimeDims[0]
+		if timeDim.Compute.TimeFloor.Grain == metricsview.TimeGrainUnspecified {
+			return nil, fmt.Errorf("time grain is required for null fill")
+		}
+
+		tz, err := time.LoadLocation(qry.TimeZone)
+		if err != nil {
+			return nil, fmt.Errorf("invalid time zone %q: %w", qry.TimeZone, err)
+		}
+
+		qry.Spine = &metricsview.Spine{}
+		s := timeutil.TruncateTime(q.TimeRange.Start.AsTime(), timeDim.Compute.TimeFloor.Grain.ToTimeutil(), tz, 1, 1)
+		e := q.TimeRange.End.AsTime()
+		qry.Spine.TimeRange = &metricsview.TimeSpine{
+			Start: s,
+			End:   e,
+			Grain: timeDim.Compute.TimeFloor.Grain,
+			Alias: timeDim.Name,
```

Comment:
I think it should be handled inside the `metricsview` package by applying it to any dimension that has a `TimeFloor` applied (and erroring if zero or multiple dimensions have a `TimeFloor`)